widgets/src/main/java/org/pentaho/gwt/widgets/client/listbox/CustomListBox.java

```diff
@@ -912,10 +912,16 @@ public void onFocus( Widget widget ) {
     if ( isEnabled() == false ) {
       return;
     }
+    if ( visible == 1 ) { // drop-down mode
+      this.togglePopup();
+    }
     // fPanel.setFocus(true);
   }
 
   public void onLostFocus( Widget widget ) {
+    if( visible == 1 ){
+      popup.hide();
+    }
```

Comment:
While the ARIA combo-box pattern also admits the possibility of opening the dropdown when it receives focus, this is not, afaik, the default HTML behavior. In my system (MacOS), up, down or space, all open the popup and show the selected element.